### PR TITLE
GH Action: Akka Stream perf

### PR DESCRIPTION
**Pull request by andreaTP**
The result of this CI is here:

https://github.com/akka/akka/actions/runs/1173937634

This is using self-hosted runners and running on JDK11.
The results are temporarily stored as CI artifacts.

### CODE REVIEW

File: .github/workflows/perf.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          fetch-depth: 0
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.2
+
+      - name: Run Performance tests
+        run: |
+          cp .jvmopts-perf-ci .jvmopts
+          sbt \
+            -Dakka.ci-server=true \
```
Comment (marcospereira) [inline .github/workflows/perf.yaml:26]: We can remove this property if env var `CI` is defined in the self-hosted runner.

File: .github/workflows/perf.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            -Dakka.test.timefactor=1 \
+            -Djava.security.egd=file:/dev/./urandom \
+            -Dakka.test.timefactor=1 \
```
Comment (marcospereira) [inline .github/workflows/perf.yaml:29]: I guess this is not relevant here.

File: .github/workflows/perf.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            "akka-bench-jmh/jmh:compile" \
+            "akka-bench-jmh/jmh:runMain akka.BenchRunner full -rf csv -rff GraphBuilderBenchmark.csv -p complexity=10 akka.stream.GraphBuilderBenchmark" \
+            "akka-bench-jmh/jmh:runMain akka.BenchRunner full -rf csv -rff MaterializationBenchmark.csv -p complexity=10 akka.stream.MaterializationBenchmark"
```
Comment (marcospereira) [inline .github/workflows/perf.yaml:32]:
```suggestion
            "akka-bench-jmh/Jmh/compile" \
            "akka-bench-jmh/Jmh/runMain akka.BenchRunner full -rf csv -rff GraphBuilderBenchmark.csv -p complexity=10 akka.stream.GraphBuilderBenchmark" \
            "akka-bench-jmh/Jmh/runMain akka.BenchRunner full -rf csv -rff MaterializationBenchmark.csv -p complexity=10 akka.stream.MaterializationBenchmark"
```

Just to minimize the number of deprecated sbt features we use.

File: .github/workflows/perf.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+      - name: 'Store produced csv files'
+        if: success()
+        uses: actions/upload-artifact@v2
+        with:
+          name: benchmark-results ${{ matrix.command }}
```
Comment (marcospereira) [inline .github/workflows/perf.yaml:38]: There is no ~spoon~ `matrix.command`.

**Review by marcospereira — COMMENTED** [review]
LGTM, @andreaTP. Some minor comments.

File: .github/workflows/perf.yaml (hunk 1 above)
Comment (andreaTP) [reply .github/workflows/perf.yaml:26]: removed

File: .github/workflows/perf.yaml (hunk 2 above)
Comment (andreaTP) [reply .github/workflows/perf.yaml:29]: is JMH, I guess I can remove it

File: .github/workflows/perf.yaml (hunk 3 above)
Comment (andreaTP) [reply .github/workflows/perf.yaml:32]: ok

**Comment by andreaTP** [comment]
@marcospereira ready for next round 🙂

**Pull request closed without merging**